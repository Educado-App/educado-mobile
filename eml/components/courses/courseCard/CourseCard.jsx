import { View, Pressable } from 'react-native';
import React, { useState } from 'react';
import { useNavigation } from '@react-navigation/native';
import Text from '../../../components/general/Text';
import { MaterialCommunityIcons } from '@expo/vector-icons';
import CustomProgressBar from '../../exercise/Progressbar';
import tailwindConfig from '../../../tailwind.config';
import { determineIcon, determineCategory, formatHours, checkProgressCourse} from '../../../services/utilityFunctions';
import DownloadCourseButton from './DownloadCourseButton';
import PropTypes from 'prop-types';
import { checkCourseStoredLocally } from '../../../services/StorageService';

/**
 * CourseCard component displays a card for a course with its details
 * @param {Object} props - Component props
 * @param {Object} props.course - Course object containing course details
 * @returns {JSX.Element} - Rendered component
 */
export default function CourseCard({ course, isOnline}) {
	const [downloaded, setDownloaded] = useState(false);
	const navigation = useNavigation();
	const [studentProgress, setStudentProgress] = useState(0);


	const checkDownload = async () => {
		setDownloaded(await checkCourseStoredLocally(course.courseId));
	};
	checkDownload();

	const checkProgress = async () => {
		const progress = await checkProgressCourse(course.courseId);
		setStudentProgress(progress);
	}; checkProgress();

	const enabledUI = 'bg-projectWhite m-[3%] rounded-lg shadow-sm shadow-opacity-[0.3] elevation-[8] mx-[5%] p-[5%]';
	const disabledUI = 'opacity-50 bg-projectWhite m-[3%] rounded-lg shadow-sm shadow-opacity-[0.3] elevation-[8] mx-[5%] p-[5%]';

	const layout = downloaded || isOnline ? enabledUI : disabledUI;

	let isDisabled = layout === disabledUI;

	return (
		<Pressable testID="courseCard"
			className={layout}
			onPress={() => { layout === enabledUI ?
				navigation.navigate('Section', {
					course: course,
				}) : null;
			}}
		>
			<View>
				<View className="flex-row items-start justify-between px-[1%] py-[1%]">
					<Text className="text-[18px] text-projectBlack flex-1 self-center font-montserrat-semi-bold">
						{course.title ? course.title : 'Título do curso'}
					</Text>
					<View className="flex-2 pr-6">
						<DownloadCourseButton course={course} disabled={isDisabled}/>
					</View>
				</View>
				<View className="h-[1] bg-disable m-[2%]" />
				<View className="flex-row flex-wrap items-center justify-start">
					<View className="flex-row items-center">
						<MaterialCommunityIcons size={18} name={determineIcon(course.category)} color={'gray'}></MaterialCommunityIcons>
						<Text className="mx-[2.5%] my-[3%]">{determineCategory(course.category)}</Text>
					</View>
					<View className="flex-row items-center">
						<MaterialCommunityIcons size={18} name="clock" color={'gray'}></MaterialCommunityIcons>
						<Text className="mx-[2.5%] my-[3%]">{course.estimatedHours ? formatHours(course.estimatedHours) : 'duração'}</Text>
					</View>
				</View>
				<View className="flex-row items-center">
					<CustomProgressBar width={56} progress={studentProgress} height={1} />
					<Pressable className="z-[1]"
						onPress={() => {layout === enabledUI ?
							navigation.navigate('Section', {
								course: course,
							}) : null;
						}}
					>
						<MaterialCommunityIcons size={28} name="play-circle" color={tailwindConfig.theme.colors.primary}></MaterialCommunityIcons>
					</Pressable>
				</View>
			</View>
		</Pressable>
	);
}

CourseCard.propTypes = {
	course: PropTypes.oneOfType([
		PropTypes.object,
		PropTypes.array,
	]),
	isOnline: PropTypes.bool
};

