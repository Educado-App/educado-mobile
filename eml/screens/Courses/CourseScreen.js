import React from 'react'
import { View, StyleSheet } from 'react-native'
import CourseTitleIcon from '../../components/courses/courseHeader/CourseTitleIcon'
import CourseProgress from '../../components/courses/courseHeader/CourseProgress'
import CourseMenu from '../../components/courses/courseHeader/CourseMenu'

export default function CourseScreen() {
  const color = '#006622'
  const type = 'material-community'
  const name = 'cash'

  return (
    <View style={styles.container}>
      <View>
        <CourseMenu />
      </View>
      <View>
        <CourseTitleIcon
          color={color}
          name={name}
          type={type}
          title={'Personal Finance'}
        ></CourseTitleIcon>
      </View>
      <CourseProgress fracTop={45} fracBot={100}></CourseProgress>
    </View>
  )
}

const styles = StyleSheet.create({})
