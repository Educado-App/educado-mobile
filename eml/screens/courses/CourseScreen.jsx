import {useNavigation} from '@react-navigation/native';
import React, {useEffect, useState} from 'react';
import {Image, Pressable, RefreshControl, ScrollView, View} from 'react-native';
import Text from '../../components/general/Text';
import * as StorageService from '../../services/StorageService';
import CourseCard from '../../components/courses/courseCard/CourseCard';
import BaseScreen from '../../components/general/BaseScreen';
import IconHeader from '../../components/general/IconHeader';
import {shouldUpdate} from '../../services/utilityFunctions';
import ToastNotification from '../../components/general/ToastNotification';
import LoadingScreen from '../../components/loading/Loading';
import NetworkStatusObserver from '../../hooks/NetworkStatusObserver';
import AsyncStorage from '@react-native-async-storage/async-storage';
import errorSwitch from '../../components/general/errorSwitch';
import ShowAlert from '../../components/general/ShowAlert';

/**
 * Course screen component that displays a list of courses.
 * @component
 * @returns {JSX.Element} The course screen component.
 */
export default function CourseScreen() {
	const [courses, setCourses] = useState([]);
	const [courseLoaded, setCourseLoaded] = useState(false);
	const [refreshing, setRefreshing] = useState(false);
	const [isOnline, setIsOnline] = useState(false);
	const [loading, setLoading] = useState(true);
	const navigation = useNavigation();

	/**
    * Asynchronous function that loads the courses from storage and updates the state.
    * @returns {void}
    */
	async function loadCourses() {
		const courseData = await StorageService.getSubCourseList();
		if (shouldUpdate(courses, courseData)) {
			if (courseData.length !== 0 && Array.isArray(courseData)) {
				setCourses(courseData);
				setCourseLoaded(true);
			}
			else {
				setCourses([]);
				setCourseLoaded(false);
			}
		}
		setLoading(false);
	}

	// When refreshing the loadCourses function is called
	const onRefresh = () => {
		setRefreshing(true);
		loadCourses();
		setRefreshing(false);
	};

	useEffect(() => {
		// this makes sure loadCourses is called when the screen is focused
		return navigation.addListener('focus', () => {
			loadCourses();
		});
	}, [navigation]);

	useEffect(() => {
		const logged = async () => {
			const loggedIn = await AsyncStorage.getItem('loggedIn');
			if (loggedIn) {
				setTimeout(async () => {
					ToastNotification('success', 'Logado!');
					await AsyncStorage.removeItem('loggedIn');
				}, 1000);
			}
		};
		try {
			logged();
		} catch (e) {
			ShowAlert(errorSwitch(e));
		}
	}, []);

	return (
		loading ? <LoadingScreen /> :
			<>
				<NetworkStatusObserver setIsOnline={setIsOnline}/>
				<BaseScreen>
					{/** Checks if the course(s) has been loaded
           * If it has, it will render and map the courses
           * If not, it will render a message saying that there are no active courses (in portugese)
           */}
					{courseLoaded ?
						<View height="100%">
							<IconHeader
								title={'Bem Vindo!'}
								description={'Aqui você encontra todos os cursos em que você está inscrito!'}/>
							<ScrollView showsVerticalScrollIndicator={false}
								refreshControl={<RefreshControl refreshing={refreshing} onRefresh={onRefresh}/>}>
								{courses.map((course, index) => (
									<CourseCard key={index} course={course} isOnline={isOnline}></CourseCard>
								)
								)}
							</ScrollView>
						</View>
						:
						<View className="bg-secondary justify-center items-center ">
							<View className="pt-24 pb-16">
								<Image source={require('../../assets/images/logo.png')} className=" justify-center items-center"/>
							</View>
							<View className=" justify-center items-center py-10 gap-10 ">
								<View className=" justify-center items-center w-full h-auto  px-10">
									{/* No active courses */}
									<Image source={require('../../assets/images/no-courses.png')}/>
									<Text
										className=" leading-[29.26] text-projectBlack pb-4 pt-4 font-sans-bold text-subheading text-center ">Comece
                      agora</Text>
									<Text className=" text-projectBlack font-montserrat text-center text-body "> Você ainda não se
                      increveu em nenhum curso. Acesse a página Explore e use a busca para encontrar cursos do seu
                      intresse.</Text>
									{/*You haven't signed up for any course yet. Access the Explore page and use the search to find courses that interest you.*/}
								</View>
								<View>
									<Pressable
										testID={'exploreButton'}
										className=" rounded-r-8 rounded-md bg-primary justify-center items-center py-4 w-full h-auto px-20 "
										onPress={() => navigation.navigate('Explorar')}>
										{/* Click to explore courses */}
										<Text className=" text-projectWhite font-sans-bold text-center text-body "> Explorar cursos</Text>
									</Pressable>
								</View>
							</View>
						</View>}
				</BaseScreen>
			</>
	);
}
