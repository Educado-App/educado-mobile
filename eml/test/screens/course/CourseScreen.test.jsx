/**
 * This file contains the test suite for the CourseScreen component.
 * It imports the necessary dependencies and mocks the useNavigation hook from react-navigation/native.
 * The test suite includes three test cases:
 * 1. Test case to check if the CourseScreen component renders correctly when there are no courses.
 * 2. Test case to check if the CourseScreen component renders correctly when courses are loaded.
 * 3. Test case to check if the CourseScreen component navigates to the explorer screen when the explore button is pressed.
 */
import React from "react";
import renderer from "react-test-renderer";
import CourseScreen from "../../../screens/courses/CourseScreen";
import CourseCard from "../../../components/courses/courseCard/CourseCard";
import AsyncStorage from "@react-native-async-storage/async-storage";

let navigated = false;
let mockNavigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
};

// Mocking the useNavigation hook from react-navigation/native
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    mockNavigation,
    navigate: jest.fn(() => { navigated = true }),
  }),
}));

/**
 * Test suite for the CourseScreen component.
 */
describe("CourseScreen", () => {
  let courseScreen;

  beforeEach(() => {
    navigated = false;
    AsyncStorage.clear();
    courseScreen = renderer.create(<CourseScreen />);
  });

  afterAll(() => {
    jest.resetModules();
    jest.restoreAllMocks();
  });

  /**
   * Test case to check if the CourseScreen component renders correctly when there are no courses.
   */
  it("renders CourseScreen correctly when theres no courses", async () => {
    expect(await courseScreen.toJSON()).toMatchSnapshot();
  });

  /**
   * Test case to check if the CourseScreen component renders correctly when courses are loaded.
   */
  it("renders CourseScreen with courses loaded", async () => {
    /**
     * Array of courses used for testing CourseScreen component.
     * @type {Array<Object>}
     */
    const courses = [
      {
        id: 1,
        title: "Course 1",
        category: "Category 1",
        duration: "Duration for course 1",
        image: "",
      },
      {
        id: 2,
        title: "Course 2",
        category: "Category 2",
        duration: "Duration for course 2",
        image: "",
      },
    ];
    const StorageService = require("../../../services/StorageService");
    jest.spyOn(StorageService, "getSubCourseList").mockResolvedValue(courses);

    await renderer.act(async () => {
      return courseScreen = renderer.create(<CourseScreen />);
    });

    expect(await courseScreen.toJSON()).toMatchSnapshot();
    expect(await courseScreen.root.findAllByType(CourseCard)).toHaveLength(2);
  });

  /**
   * Test case to check if the CourseScreen component navigates to the explorer screen when the explore button is pressed.
   */
  it('Navigate to explorer when pressing the explore button', async () => {
    const button = await courseScreen.root.findByProps({ testID: 'exploreButton' });
    button.props.onPress();
    expect(navigated).toBe(true);
  });
});