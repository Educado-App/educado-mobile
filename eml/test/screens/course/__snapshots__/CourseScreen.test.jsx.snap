// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseScreen renders CourseScreen correctly when theres no courses 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#F1F9FB",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "justifyContent": "center",
        },
        {
          "alignItems": "center",
        },
        {
          "backgroundColor": "#F1F9FB",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "paddingTop": 96,
          },
          {
            "paddingBottom": 64,
          },
        ]
      }
    >
      <Image
        source={1}
        style={
          [
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "width": 175.88,
            },
            {
              "height": 25.54,
            },
          ]
        }
      />
    </View>
    <View
      style={
        [
          {
            "justifyContent": "center",
          },
          {
            "alignItems": "center",
          },
          {
            "paddingBottom": 96,
          },
          {
            "paddingTop": 96,
          },
          {
            "marginLeft": -40,
            "marginTop": -40,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "justifyContent": "center",
            },
            {
              "alignItems": "center",
            },
            {
              "width": 342,
            },
            {
              "height": 308.02,
            },
            [
              {
                "marginLeft": 40,
                "marginTop": 40,
              },
            ],
          ]
        }
      >
        <Image
          source={1}
        />
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              [
                {
                  "color": "#383838",
                },
                {
                  "paddingBottom": 16,
                },
                {
                  "paddingTop": 16,
                },
                {
                  "fontFamily": "Montserrat-Bold",
                },
                {
                  "fontSize": 24,
                },
                {
                  "textAlign": "center",
                },
              ],
            ]
          }
        >
          Comece agora
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Montserrat-Regular",
              },
              [
                {
                  "color": "#383838",
                },
                {
                  "fontFamily": "Montserrat-Regular",
                },
                {
                  "textAlign": "center",
                },
                {
                  "fontSize": 16,
                },
              ],
            ]
          }
        >
           Você ainda não se increveu em nenhum curso. Acesse a página Explore e use a busca para encontrar cursos do seu intresse.
        </Text>
      </View>
      <View
        style={
          [
            {
              "marginLeft": 40,
              "marginTop": 40,
            },
          ]
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "borderBottomLeftRadius": 6,
                "borderBottomRightRadius": 6,
                "borderTopLeftRadius": 6,
                "borderTopRightRadius": 6,
              },
              {
                "backgroundColor": "#5ECCE9",
              },
              {
                "justifyContent": "center",
              },
              {
                "alignItems": "center",
              },
              {
                "paddingBottom": 8,
                "paddingLeft": 8,
                "paddingRight": 8,
                "paddingTop": 8,
              },
              {
                "height": 52,
              },
              {
                "width": 342,
              },
            ]
          }
          testID="exploreButton"
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Montserrat-Regular",
                },
                [
                  {
                    "color": "#FFFFFF",
                  },
                  {
                    "fontFamily": "Montserrat-Bold",
                  },
                  {
                    "textAlign": "center",
                  },
                  {
                    "fontSize": 16,
                  },
                ],
              ]
            }
          >
             Explore courses
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`CourseScreen renders CourseScreen with courses loaded 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#F1F9FB",
      },
    ]
  }
>
  <View
    height="100%"
  >
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "paddingLeft": 24,
          },
          {
            "paddingTop": "20%",
          },
          {
            "paddingBottom": "10%",
          },
        ]
      }
    >
      <Image
        alt="Icon"
        source={1}
        style={
          [
            {
              "width": 32,
            },
            {
              "height": 32,
            },
            {
              "marginRight": 8,
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "fontFamily": "Montserrat-Regular",
            },
            [
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              {
                "fontWeight": "700",
              },
            ],
          ]
        }
      >
        Bem Vindo!
      </Text>
    </View>
    <RCTScrollView
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "marginBottom": "3%",
                "marginLeft": "3%",
                "marginRight": "3%",
                "marginTop": "3%",
              },
              {
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
              },
              {
                "elevation": 8,
              },
              {
                "paddingBottom": "5%",
                "paddingLeft": "5%",
                "paddingRight": "5%",
                "paddingTop": "5%",
              },
            ]
          }
          testID="courseCard"
        >
          <View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "flex-start",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                  {
                    "paddingLeft": "1%",
                    "paddingRight": "1%",
                  },
                  {
                    "paddingBottom": "1%",
                    "paddingTop": "1%",
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": undefined,
                      "fontSize": 28,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰑴
                 
              </Text>
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "fontSize": 18,
                      },
                      {
                        "flexBasis": "0%",
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                      {
                        "alignSelf": "center",
                      },
                    ],
                  ]
                }
              >
                Course 1
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "height": 1,
                  },
                  {
                    "backgroundColor": "#e0e0e0",
                  },
                  {
                    "marginBottom": "2%",
                    "marginLeft": "2%",
                    "marginRight": "2%",
                    "marginTop": "2%",
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                  {
                    "justifyContent": "flex-start",
                  },
                  {
                    "overflow": "hidden",
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "gray",
                      "fontSize": 18,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰑴
              </Text>
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "marginLeft": "2.5%",
                        "marginRight": "2.5%",
                      },
                      {
                        "marginBottom": "3%",
                        "marginTop": "3%",
                      },
                    ],
                  ]
                }
              >
                Category 1
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "gray",
                      "fontSize": 18,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰥔
              </Text>
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "marginLeft": "2.5%",
                        "marginRight": "2.5%",
                      },
                      {
                        "marginBottom": "3%",
                        "marginTop": "3%",
                      },
                    ],
                  ]
                }
              >
                duration
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    {
                      "alignItems": "center",
                    },
                    {
                      "justifyContent": "space-around",
                    },
                  ]
                }
              >
                <View
                  animated={true}
                  animationConfig={
                    {
                      "bounciness": 0,
                    }
                  }
                  animationType="spring"
                  indeterminate={false}
                  indeterminateAnimationDuration={1000}
                  onLayout={[Function]}
                  progress={0.5}
                  style={
                    [
                      {
                        "backgroundColor": "rgba(228, 242, 245, 1)",
                        "borderColor": "rgba(94, 204, 233, 1)",
                        "borderRadius": 8,
                        "borderWidth": 0,
                        "overflow": "hidden",
                        "width": 450,
                      },
                      undefined,
                    ]
                  }
                  useNativeDriver={false}
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "backgroundColor": "rgba(94, 204, 233, 1)",
                        "height": 13.34,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateX": -112.5,
                          },
                          {
                            "scaleX": 0.50005,
                          },
                        ],
                      }
                    }
                  />
                </View>
                <Text
                  style={
                    [
                      {
                        "paddingLeft": 20,
                        "paddingRight": 20,
                      },
                      {
                        "textAlign": "center",
                      },
                      {
                        "fontFamily": "Montserrat-Bold",
                      },
                      {
                        "fontSize": 12,
                      },
                      {
                        "color": "#383838",
                      },
                    ]
                  }
                >
                  50
                  %
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "zIndex": 1,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#5ECCE9",
                        "fontSize": 28,
                      },
                      undefined,
                      {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰐌
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "backgroundColor": "#fff",
              },
              {
                "marginBottom": "3%",
                "marginLeft": "3%",
                "marginRight": "3%",
                "marginTop": "3%",
              },
              {
                "borderBottomLeftRadius": 10,
                "borderBottomRightRadius": 10,
                "borderTopLeftRadius": 10,
                "borderTopRightRadius": 10,
              },
              {
                "elevation": 8,
              },
              {
                "paddingBottom": "5%",
                "paddingLeft": "5%",
                "paddingRight": "5%",
                "paddingTop": "5%",
              },
            ]
          }
          testID="courseCard"
        >
          <View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "flex-start",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                  {
                    "paddingLeft": "1%",
                    "paddingRight": "1%",
                  },
                  {
                    "paddingBottom": "1%",
                    "paddingTop": "1%",
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": undefined,
                      "fontSize": 28,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰑴
                 
              </Text>
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "fontSize": 18,
                      },
                      {
                        "flexBasis": "0%",
                        "flexGrow": 1,
                        "flexShrink": 1,
                      },
                      {
                        "alignSelf": "center",
                      },
                    ],
                  ]
                }
              >
                Course 2
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "height": 1,
                  },
                  {
                    "backgroundColor": "#e0e0e0",
                  },
                  {
                    "marginBottom": "2%",
                    "marginLeft": "2%",
                    "marginRight": "2%",
                    "marginTop": "2%",
                  },
                ]
              }
            />
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                  {
                    "justifyContent": "flex-start",
                  },
                  {
                    "overflow": "hidden",
                  },
                ]
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "gray",
                      "fontSize": 18,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰑴
              </Text>
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "marginLeft": "2.5%",
                        "marginRight": "2.5%",
                      },
                      {
                        "marginBottom": "3%",
                        "marginTop": "3%",
                      },
                    ],
                  ]
                }
              >
                Category 2
              </Text>
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  [
                    {
                      "color": "gray",
                      "fontSize": 18,
                    },
                    undefined,
                    {
                      "fontFamily": "material-community",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    {},
                  ]
                }
              >
                󰥔
              </Text>
              <Text
                style={
                  [
                    {
                      "fontFamily": "Montserrat-Regular",
                    },
                    [
                      {
                        "marginLeft": "2.5%",
                        "marginRight": "2.5%",
                      },
                      {
                        "marginBottom": "3%",
                        "marginTop": "3%",
                      },
                    ],
                  ]
                }
              >
                duration
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "flexDirection": "row",
                    },
                    {
                      "alignItems": "center",
                    },
                    {
                      "justifyContent": "space-around",
                    },
                  ]
                }
              >
                <View
                  animated={true}
                  animationConfig={
                    {
                      "bounciness": 0,
                    }
                  }
                  animationType="spring"
                  indeterminate={false}
                  indeterminateAnimationDuration={1000}
                  onLayout={[Function]}
                  progress={0.5}
                  style={
                    [
                      {
                        "backgroundColor": "rgba(228, 242, 245, 1)",
                        "borderColor": "rgba(94, 204, 233, 1)",
                        "borderRadius": 8,
                        "borderWidth": 0,
                        "overflow": "hidden",
                        "width": 450,
                      },
                      undefined,
                    ]
                  }
                  useNativeDriver={false}
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "backgroundColor": "rgba(94, 204, 233, 1)",
                        "height": 13.34,
                        "transform": [
                          {
                            "translateX": 0,
                          },
                          {
                            "translateX": -112.5,
                          },
                          {
                            "scaleX": 0.50005,
                          },
                        ],
                      }
                    }
                  />
                </View>
                <Text
                  style={
                    [
                      {
                        "paddingLeft": 20,
                        "paddingRight": 20,
                      },
                      {
                        "textAlign": "center",
                      },
                      {
                        "fontFamily": "Montserrat-Bold",
                      },
                      {
                        "fontSize": 12,
                      },
                      {
                        "color": "#383838",
                      },
                    ]
                  }
                >
                  50
                  %
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "zIndex": 1,
                    },
                  ]
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    [
                      {
                        "color": "#5ECCE9",
                        "fontSize": 28,
                      },
                      undefined,
                      {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰐌
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
